// ng-angular-popup/esm2022/lib/ng-toast.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgToastComponent } from './ng-toast.component';
import { ToastIconDirective } from './toast-icon.directive';
import * as i0 from "@angular/core";
export class NgToastModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.0.1", ngImport: i0, type: NgToastModule, declarations: [NgToastComponent, ToastIconDirective], imports: [CommonModule], exports: [NgToastComponent, ToastIconDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NgToastComponent, ToastIconDirective],
                    imports: [CommonModule],
                    exports: [NgToastComponent, ToastIconDirective],
                }]
        }] });
