// ng-angular-popup/esm2022/lib/ng-toast.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { ToastMessage } from './toast-message.model';
import { ToastType } from './toast-type.enum';
import * as i0 from "@angular/core";
/**
 * Service for displaying toast messages.
 */
export class NgToastService {
    #defaultDuration;
    #toastMessageSource;
    /**
     * Constructs a new NgToastService instance.
     */
    constructor() {
        this.#defaultDuration = 2000;
        this.#toastMessageSource = new Subject();
    }
    /**
     * Displays a toast message.
     * @param message The message to display.
     * @param type The type of the toast message.
     * @param title The optional title of the toast message.
     * @param duration The duration in milliseconds for which the toast message should be displayed. Defaults to the default duration.
     */
    toast(message, type, title, duration = this.#defaultDuration) {
        this.#toastMessageSource.next(new ToastMessage(message, type, title, duration));
    }
    /**
     * Displays a success toast message.
     * @param message The message to display.
     * @param title The optional title of the toast message.
     * @param duration The duration in milliseconds for which the toast message should be displayed. Defaults to the default duration.
     */
    success(message, title, duration = this.#defaultDuration) {
        this.toast(message, ToastType.SUCCESS, title, duration);
    }
    /**
     * Displays an info toast message.
     * @param message The message to display.
     * @param title The optional title of the toast message.
     * @param duration The duration in milliseconds for which the toast message should be displayed. Defaults to the default duration.
     */
    info(message, title, duration = this.#defaultDuration) {
        this.toast(message, ToastType.INFO, title, duration);
    }
    /**
     * Displays a warning toast message.
     * @param message The message to display.
     * @param title The optional title of the toast message.
     * @param duration The duration in milliseconds for which the toast message should be displayed. Defaults to the default duration.
     */
    warning(message, title, duration = this.#defaultDuration) {
        this.toast(message, ToastType.WARNING, title, duration);
    }
    /**
     * Displays a danger/error toast message.
     * @param message The message to display.
     * @param title The optional title of the toast message.
     * @param duration The duration in milliseconds for which the toast message should be displayed. Defaults to the default duration.
     */
    danger(message, title, duration = this.#defaultDuration) {
        this.toast(message, ToastType.DANGER, title, duration);
    }
    /**
     * Returns an observable that emits the toast messages.
     * @returns An observable that emits the toast messages.
     */
    onToastMessage() {
        return this.#toastMessageSource.asObservable();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: NgToastService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [] });
